support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/response/accesstoken/OAuth20DefaultTokenGenerator.java

```diff
@@ -205,10 +205,10 @@ private OAuth20DeviceToken getDeviceTokenFromTicketRegistry(final String deviceC
         addTicketToRegistry(accessToken, ticketGrantingTicket);
         LOGGER.debug("Added access token [{}] to registry", accessToken);
 
-        updateOAuthCode(holder);
+        updateOAuthCode(holder, accessToken.getId());
 
         val refreshToken = FunctionUtils.doIf(holder.isGenerateRefreshToken(),
-            () -> generateRefreshToken(holder),
+            () -> generateRefreshToken(holder, accessToken.getId()),
```

Comment:
Pass the access token, and the use the id in the callee.

Reply:
Done